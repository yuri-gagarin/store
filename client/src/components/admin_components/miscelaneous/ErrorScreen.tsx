import React from 'react'
import { Button, Dimmer, Header, Icon, Segment } from 'semantic-ui-react'
import axios from "axios";

type RetryLastReqFunc = <T> () => T
interface Props {
  errorMessage?: string;
  lastRequest?: () => Promise<void>;
}
const ErrorScreen: React.FC<Props> = ({ errorMessage, lastRequest }): JSX.Element => {

  const handleRetry = (e: React.MouseEvent<HTMLButtonElement>): void => {
    if (lastRequest) {
      lastRequest();
    }
  };
  
  return (
    <div style={{ height: "100vh" }}> 
      <Segment style={{ height: "100vh" }}>
        <Dimmer active>
          <Header as='h2' icon inverted>
            <Icon name='exclamation circle' />
              An Error Occured!
            <Header.Subheader>Dimmer sub-header</Header.Subheader>
            <Button 
              id="errorScreenRetryButton"
              inverted 
              color="green" 
              content="Retry" 
              onClick={handleRetry} 
            />
          </Header>
        </Dimmer>
      </Segment>
    </div>
  )
};

export default ErrorScreen