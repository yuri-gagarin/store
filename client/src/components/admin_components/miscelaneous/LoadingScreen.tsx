import React from 'react'
import { Dimmer, Loader, Segment } from 'semantic-ui-react'

const LoaderExampleText: React.FC<{}> = (props): JSX.Element => {
  return (
    <div style={{ height: "100vh" }}> 
      <Segment style={{ height: "100vh" }}>
        <Dimmer active inverted>
          <Loader>Please Wait</Loader>
        </Dimmer>
      </Segment>
    </div>
  )
};

export default LoaderExampleText