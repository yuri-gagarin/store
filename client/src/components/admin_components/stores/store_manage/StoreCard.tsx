import React, { useState } from "react";
import { Button, Grid } from "semantic-ui-react";
import { IGlobalAppState, AppAction } from "../../../../state/Store";
import { deleteStore, getStore } from "../actions/APIstoreActions";
import { setCurrentStore, clearCurrentStore } from "../actions/uiStoreActions";
import { withRouter, RouteComponentProps } from "react-router-dom";
// css imports //
import "./css/storeCard.css";
import { ConvertDate } from "../../../helpers/displayHelpers";

interface StoreCardProps extends RouteComponentProps {
  _id: string;
  imgUrl?: string;
  title: string;
  description: string;
  imageCount: number;
  createdAt: string;
  editedAt?: string;
  state: IGlobalAppState;
  dispatch: React.Dispatch<AppAction>;
}
type EditControlsProps = {
  handleStoreEdit(e:  React.MouseEvent<HTMLButtonElement>): void;
}

const CancelEditControls: React.FC<EditControlsProps> = ({ handleStoreEdit }): JSX.Element => {
  return (
    <Grid.Row>
      <Grid.Column>
         <Button content="Close Edit" onClick={handleStoreEdit}></Button>
      </Grid.Column>
    </Grid.Row>
  )
}

const StoreCard: React.FC<StoreCardProps> = ({ 
   _id, title, description, imageCount, createdAt, editedAt, history,
   state, dispatch
  }): JSX.Element => {
  const [ editing, setEditing ] = useState<boolean>(false);
  const baseUrl = "/admin/home/my_store/manage"
 
  const handleStoreOpen = (e: React.MouseEvent<HTMLButtonElement>): void => {
    history.push(baseUrl + "/view");
  }
  const handleStoreEdit = (e: React.MouseEvent<HTMLButtonElement>): void => {
    if (!editing) {
      //setCurrentStore(_id, dispatch, state);
      //history.push(baseUrl + "/edit");
      //setEditing(true);

      
      getStore(_id, dispatch)
        .then((success) => {
          if (success) {
            history.push(baseUrl + "/edit");
            setEditing(true);
          }
        
        });
      
     
    } else {
      clearCurrentStore(dispatch);
      history.push(baseUrl);
      setEditing(false);
    }
  }
  const handleStoreDelete = (e: React.MouseEvent<HTMLButtonElement>): void => {
    deleteStore( _id, dispatch, state)
      .then((success) => {
        
      })
  }

  return (
    <React.Fragment>
      <Grid.Row style={{ border: "3px solid green", padding: "0.5em", marginTop: "2em" }}>
        <Grid.Column computer={12} mobile={16} style={{ border: '2px solid red'}}>
          <h3>Title: {title}</h3>
          <h5>Description: {description}</h5>
          <h5>Number of Images: {imageCount}</h5>
          <p><span>Created At: </span>{ConvertDate.international(createdAt)}</p>
        </Grid.Column> 
        <Grid.Column computer={4} mobile={16} style={{ border: '2px solid blue'}}>
          <Button 
            inverted
            color="green"
            className="storeCardBtn" 
            content="Open" 
            onClick={handleStoreOpen} 
          />
          <Button 
            inverted
            color="orange"
            className="storeCardBtn" 
            content="Edit" 
            onClick={handleStoreEdit}  
          />
          <Button 
            inverted
            color="red"
            className="storeCardBtn" 
            content="Delete" 
            onClick={handleStoreDelete}
          />
        </Grid.Column> 
      </Grid.Row>
      {  editing ?  <CancelEditControls handleStoreEdit={handleStoreEdit}/>: null }
    </React.Fragment>
    
  );
};

export default withRouter(StoreCard);