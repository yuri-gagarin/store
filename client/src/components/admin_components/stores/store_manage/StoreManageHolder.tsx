import React, { useContext, useEffect } from "react";
import { Button, Grid } from "semantic-ui-react";
import StoreFormHolder from "../forms/StoreFormHolder";
import StoreCard from "./StoreCard";
import { withRouter, RouteComponentProps, useRouteMatch, Route } from "react-router-dom";
import { Store } from "../../../../state/Store";
import { getAllStores } from "../actions/APIstoreActions";

interface Props extends RouteComponentProps {};

const StoreManageHolder: React.FC<Props> = ({ history }): JSX.Element => {
  const { state, dispatch } = useContext(Store);
  const { loadedStores } = state.storeState;
  const match = useRouteMatch("/admin/home/my_store/manage");

  const handleBack = () => {
    history.goBack();
  }
  useEffect(() => {
    getAllStores(dispatch);
  }, []); 

  return (
    <Grid padded>
       
      <Route path={match?.url + "/edit"}> 
        <Grid.Row>
          <Grid.Column>
            <h3>Editing Store: { state.storeState.currentStoreData.title }</h3>
            <Button inverted color="green" content="Back" onClick={handleBack}></Button>
          </Grid.Column>
        </Grid.Row>
        <StoreFormHolder state={state} dispatch={dispatch} />
      </Route>
      <Route exact path={match?.url}>
        {
          loadedStores.map((store) => {
            return (
              <StoreCard 
                key={store._id}
                _id={store._id}
                title={store.title}
                description={store.description}
                imageCount={store.images.length}
                createdAt={store.createdAt}
                editedAt={store.editedAt}
                state={state}
                dispatch={dispatch}
              />
            );
          })
        }
      </Route>
     
    </Grid>
  );
};

export default withRouter(StoreManageHolder);