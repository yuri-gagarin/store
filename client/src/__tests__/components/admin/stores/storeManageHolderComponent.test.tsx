import React from "react";
import { Grid } from "semantic-ui-react";
// test dependencies //
import moxios from "moxios";
import { mount, ReactWrapper } from "enzyme";
import { act } from "react-dom/test-utils";
import { MemoryRouter as Router } from "react-router-dom";
// components //
import StoreManageHolder from "../../../../components/admin_components/stores/store_manage/StoreManageHolder";
import LoadingScreen from "../../../../components/admin_components/miscelaneous/LoadingScreen";
import StoreCard from "../../../../components/admin_components/stores/store_manage/StoreCard";
import ErrorScreen from "../../../../components/admin_components/miscelaneous/ErrorScreen";
// helpers and state //
import { TestStateProvider } from "../../../../state/Store";
import { AdminStoreRoutes } from "../../../../routes/adminRoutes";

describe("Store Manage Holder Tests", () => {
    let mockStores: IStoreData[];

    beforeAll(() => {
      mockStores = [
        {
          _id: "1",
          title: "first",
          description: "desc",
          images: [],
          createdAt: "now"
        },
        {
          _id: "2",
          title: "second",
          description: "desc",
          images: [],
          createdAt: "now"
        }
      ]
    })
    describe("Default Component state at first render", () => {
      let wrapper: ReactWrapper; 

      beforeAll( async () => {
        moxios.install();
        wrapper = mount(
          <Router keyLength={0} initialEntries={[AdminStoreRoutes.MANAGE_ROUTE]}>
            <TestStateProvider>
              <StoreManageHolder />
            </TestStateProvider>
          </Router>
        );

        await act( async () => {
          await moxios.stubRequest("/api/stores", {
            status: 200,
            response: {
              responseMsg: "All Ok",
              stores: []
            }
          });
        });

      });
       afterAll(() => {
        moxios.uninstall();
      }); 
      
      it("Should correctly render", () => {
        expect(wrapper.find(StoreManageHolder)).toMatchSnapshot();
      });
      
      it("Should render a 'LoadingScreen' Component before an API call resolves", () => {
        const loadingScreen = wrapper.find(LoadingScreen);
        expect(loadingScreen.length).toEqual(1)
      });
    });
    
    // mock successful API call render tests //
    describe("State after a successful API call", () => {
      let wrapper: ReactWrapper; let loadingScreen: ReactWrapper;

      beforeAll( async () => {
        moxios.install();

        wrapper = mount(
          <Router keyLength={0} initialEntries={[AdminStoreRoutes.MANAGE_ROUTE]}>
            <TestStateProvider>
              <StoreManageHolder />
            </TestStateProvider>
          </Router>
        );
        
        await act( async () => {
          await moxios.stubRequest("/api/stores", {
            status: 200,
            response: {
              responseMsg: "All Ok",
              stores: mockStores
            }
          });
        });
      });

      afterAll(() => {
        moxios.uninstall();
      }); 

      it("Should correctly render the initial Loading Screen", () => {
        const loadingScreen = wrapper.find(LoadingScreen);
        expect(loadingScreen.length).toEqual(1);
      });
      it("Should not render the initial Loading Screen after the API call", async () => {
        act( () => {
          wrapper.update();
        });
        const loadingScreen = wrapper.find(LoadingScreen);
        expect(loadingScreen.length).toEqual(0);
      });
      it("Should NOT render the ErrorScreen Component", () => {
        const errorScreenComponent = wrapper.find(ErrorScreen);
        expect(errorScreenComponent.length).toEqual(0);
      });
  
      it("Should render the correct StoreManageHolder Component", () => {
        const storeManageHolderComp = wrapper.find(StoreManageHolder).find(Grid);
        expect(storeManageHolderComp.length).toEqual(1)
        expect(storeManageHolderComp).toMatchSnapshot();
      });
    
      it("Should render correct number of StoreCard components", () => {
        const storeCards = wrapper.find(StoreCard);
        expect(storeCards.length).toEqual(mockStores.length);
      })
    });
    // END mock successfull API call render tests //
    // mock ERROR API call render tests //
    describe("State after a Error in API call", () => {
      let wrapper: ReactWrapper;

      beforeAll(async () => {
        await act(async () => {
          moxios.install();
          wrapper = await mount(
            <Router keyLength={0} initialEntries={[AdminStoreRoutes.MANAGE_ROUTE]}>
              <TestStateProvider>
                <StoreManageHolder />
              </TestStateProvider>
            </Router>
          );
          moxios.stubRequest("/api/stores", {
            status: 500,
            response: {
              responseMsg: "Error here",
              error: new Error("API Call Error")
            }
          });
        });
        act(() => {
          moxios.uninstall();
        })
      });

      it("Should correctly render the initial Loading Screen", () => {
        const loadingScreen = wrapper.find(LoadingScreen);
        expect(loadingScreen.length).toEqual(1);
      });

      it("Should not render the initial Loading Screen after an  API call", () => {
        act( () => {
          wrapper.update();
        });
        const loadingScreen = wrapper.find(LoadingScreen);
        expect(loadingScreen.length).toEqual(0);
      });
      it("Should render the ErrorScreen Component", () => {
        const errorScreenComponent = wrapper.find(ErrorScreen);
        expect(errorScreenComponent.length).toEqual(1);
      });
      
      it("Should NOT render the StoreManageHolder Component", () => {
        const storeManageHolderComp = wrapper.find(StoreManageHolder).find(Grid);
        expect(storeManageHolderComp.length).toEqual(0);
      });
    
      it("Should NOT render ANY StoreCard components", () => {
        const storeCards = wrapper.find(StoreCard);
        expect(storeCards.length).toEqual(0);
      });
  
      it("Should have a retry Store API call Button", () => {
        const retryButton = wrapper.find(ErrorScreen).render().find("#errorScreenRetryButton");
        expect(retryButton.length).toEqual(1);
      });
      
      
      it("Should correctly re-dispatch the 'getStores' API request with the button click", async () => {
        await act( async () => {
          moxios.install();
          moxios.stubRequest("/api/stores", {
            status: 200,
            response: {
              responseMsg: "All Ok",
              stores: mockStores
            }
          });
          const retryButton = wrapper.find("#errorScreenRetryButton");
          retryButton.at(0).simulate("click");
        });
      });
      it("Should properly rerender 'StoreManageHolder' component", () => {
        wrapper.update()
        const errorScreen = wrapper.find(ErrorScreen);
        const storeManageHolderComp = wrapper.find(StoreManageHolder).find(Grid);
        // assert correct rendering //
        expect(errorScreen.length).toEqual(0);
        expect(storeManageHolderComp.length).toEqual(1);
      });
      
      it("Should render correct number of 'StoreCard' Components", () => {
        const storeCards = wrapper.find(StoreCard);
        expect(storeCards.length).toEqual(mockStores.length);
      });
    });
    // END mock successfull API call tests //
    
  
});