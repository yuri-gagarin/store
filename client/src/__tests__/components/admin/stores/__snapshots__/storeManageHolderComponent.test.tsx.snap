// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store Manage Holder Tests Default Component state at first render Should correctly render 1`] = `
<withRouter(StoreManageHolder)>
  <StoreManageHolder
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/admin/home/my_stores/manage",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/admin/home/my_stores/manage",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "hash": "",
        "pathname": "/admin/home/my_stores/manage",
        "search": "",
        "state": undefined,
      }
    }
    match={
      Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      }
    }
  >
    <LoaderExampleText>
      <div
        id="generalLoadingScreen"
        style={
          Object {
            "height": "100vh",
          }
        }
      >
        <Segment
          style={
            Object {
              "height": "100vh",
            }
          }
        >
          <div
            className="ui segment"
            style={
              Object {
                "height": "100vh",
              }
            }
          >
            <Dimmer
              active={true}
              inverted={true}
            >
              <DimmerInner
                active={true}
                inverted={true}
              >
                <Ref
                  innerRef={
                    Object {
                      "current": null,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": null,
                      }
                    }
                  >
                    <div
                      className="ui active transition visible inverted dimmer"
                      onClick={[Function]}
                    >
                      <div
                        className="content"
                      >
                        <Loader>
                          <div
                            className="ui text loader"
                          >
                            Please Wait
                          </div>
                        </Loader>
                      </div>
                    </div>
                  </RefFindNode>
                </Ref>
              </DimmerInner>
            </Dimmer>
          </div>
        </Segment>
      </div>
    </LoaderExampleText>
  </StoreManageHolder>
</withRouter(StoreManageHolder)>
`;

exports[`Store Manage Holder Tests State after a successful API call Should render the correct StoreManageHolder Component 1`] = `
<Grid
  id="storeManageHolder"
  padded={true}
>
  <div
    className="ui padded grid"
    id="storeManageHolder"
  >
    <Route
      path="/admin/home/my_stores/manage/edit"
    />
    <Route
      exact={true}
      path="/admin/home/my_stores/manage"
    >
      <withRouter(StoreCard)
        _id="1"
        createdAt="now"
        description="desc"
        dispatch={[Function]}
        imageCount={0}
        key="1"
        state={
          Object {
            "bonusVideoState": Object {
              "bonusVideoFormOpen": false,
              "currentBonusVideoData": Object {
                "_id": "",
                "createdAt": "",
                "description": "",
                "vimeoURL": "",
                "youTubeURL": "",
              },
              "error": null,
              "loadedBonusVideos": Array [],
              "loading": false,
              "responseMsg": "",
            },
            "productState": Object {
              "currentProductData": Object {
                "_id": "",
                "createdAt": "",
                "description": "",
                "details": "",
                "images": Array [],
                "name": "",
                "price": "",
              },
              "error": null,
              "loadedProducts": Array [],
              "loading": false,
              "productFormOpen": false,
              "responseMsg": "",
            },
            "serviceState": Object {
              "currentServiceData": Object {
                "_id": "",
                "createdAt": "",
                "description": "",
                "images": Array [],
                "name": "",
                "price": "",
              },
              "error": null,
              "loadedServices": Array [],
              "loading": false,
              "responseMsg": "",
              "serviceFormOpen": false,
            },
            "storeItemState": Object {
              "currentStoreItemData": Object {
                "_id": "",
                "categories": Array [],
                "createdAt": "",
                "description": "",
                "details": "",
                "images": Array [],
                "name": "",
                "price": "",
                "storeId": "",
                "storeName": "",
              },
              "error": null,
              "loadedStoreItems": Array [],
              "loading": false,
              "numberOfItems": 0,
              "responseMsg": "",
              "storeItemFormOpen": false,
            },
            "storeState": Object {
              "currentStoreData": Object {
                "_id": "",
                "createdAt": "",
                "description": "",
                "images": Array [],
                "title": "",
              },
              "error": null,
              "loadedStores": Array [
                Object {
                  "_id": "1",
                  "createdAt": "now",
                  "description": "desc",
                  "images": Array [],
                  "title": "first",
                },
                Object {
                  "_id": "2",
                  "createdAt": "now",
                  "description": "desc",
                  "images": Array [],
                  "title": "second",
                },
              ],
              "loading": false,
              "responseMsg": "All Ok",
              "storeFormOpen": false,
            },
          }
        }
        title="first"
      >
        <StoreCard
          _id="1"
          createdAt="now"
          description="desc"
          dispatch={[Function]}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "/admin/home/my_stores/manage",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/admin/home/my_stores/manage",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          imageCount={0}
          location={
            Object {
              "hash": "",
              "pathname": "/admin/home/my_stores/manage",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/admin/home/my_stores/manage",
              "url": "/admin/home/my_stores/manage",
            }
          }
          state={
            Object {
              "bonusVideoState": Object {
                "bonusVideoFormOpen": false,
                "currentBonusVideoData": Object {
                  "_id": "",
                  "createdAt": "",
                  "description": "",
                  "vimeoURL": "",
                  "youTubeURL": "",
                },
                "error": null,
                "loadedBonusVideos": Array [],
                "loading": false,
                "responseMsg": "",
              },
              "productState": Object {
                "currentProductData": Object {
                  "_id": "",
                  "createdAt": "",
                  "description": "",
                  "details": "",
                  "images": Array [],
                  "name": "",
                  "price": "",
                },
                "error": null,
                "loadedProducts": Array [],
                "loading": false,
                "productFormOpen": false,
                "responseMsg": "",
              },
              "serviceState": Object {
                "currentServiceData": Object {
                  "_id": "",
                  "createdAt": "",
                  "description": "",
                  "images": Array [],
                  "name": "",
                  "price": "",
                },
                "error": null,
                "loadedServices": Array [],
                "loading": false,
                "responseMsg": "",
                "serviceFormOpen": false,
              },
              "storeItemState": Object {
                "currentStoreItemData": Object {
                  "_id": "",
                  "categories": Array [],
                  "createdAt": "",
                  "description": "",
                  "details": "",
                  "images": Array [],
                  "name": "",
                  "price": "",
                  "storeId": "",
                  "storeName": "",
                },
                "error": null,
                "loadedStoreItems": Array [],
                "loading": false,
                "numberOfItems": 0,
                "responseMsg": "",
                "storeItemFormOpen": false,
              },
              "storeState": Object {
                "currentStoreData": Object {
                  "_id": "",
                  "createdAt": "",
                  "description": "",
                  "images": Array [],
                  "title": "",
                },
                "error": null,
                "loadedStores": Array [
                  Object {
                    "_id": "1",
                    "createdAt": "now",
                    "description": "desc",
                    "images": Array [],
                    "title": "first",
                  },
                  Object {
                    "_id": "2",
                    "createdAt": "now",
                    "description": "desc",
                    "images": Array [],
                    "title": "second",
                  },
                ],
                "loading": false,
                "responseMsg": "All Ok",
                "storeFormOpen": false,
              },
            }
          }
          title="first"
        >
          <GridRow
            style={
              Object {
                "border": "3px solid green",
                "marginTop": "2em",
                "padding": "0.5em",
              }
            }
          >
            <div
              className="row"
              style={
                Object {
                  "border": "3px solid green",
                  "marginTop": "2em",
                  "padding": "0.5em",
                }
              }
            >
              <GridColumn
                computer={12}
                mobile={16}
                style={
                  Object {
                    "border": "2px solid red",
                  }
                }
              >
                <div
                  className="twelve wide computer sixteen wide mobile column"
                  style={
                    Object {
                      "border": "2px solid red",
                    }
                  }
                >
                  <h3>
                    Title: 
                    first
                  </h3>
                  <h5>
                    Description: 
                    desc
                  </h5>
                  <h5>
                    Number of Images: 
                    0
                  </h5>
                  <p>
                    <span>
                      Created At: 
                    </span>
                    NaN-undefined-NaN
                  </p>
                </div>
              </GridColumn>
              <GridColumn
                computer={4}
                mobile={16}
                style={
                  Object {
                    "border": "2px solid blue",
                  }
                }
              >
                <div
                  className="four wide computer sixteen wide mobile column"
                  style={
                    Object {
                      "border": "2px solid blue",
                    }
                  }
                >
                  <Button
                    as="button"
                    className="storeCardBtn"
                    color="green"
                    content="Open"
                    inverted={true}
                    onClick={[Function]}
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui green inverted button storeCardBtn"
                          >
                            Open
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui green inverted button storeCardBtn"
                            >
                              Open
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui green inverted button storeCardBtn"
                          onClick={[Function]}
                        >
                          Open
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                  <Button
                    as="button"
                    className="storeCardBtn"
                    color="orange"
                    content="Edit"
                    inverted={true}
                    onClick={[Function]}
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui orange inverted button storeCardBtn"
                          >
                            Edit
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui orange inverted button storeCardBtn"
                            >
                              Edit
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui orange inverted button storeCardBtn"
                          onClick={[Function]}
                        >
                          Edit
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                  <Button
                    as="button"
                    className="storeCardBtn"
                    color="red"
                    content="Delete"
                    inverted={true}
                    onClick={[Function]}
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui red inverted button storeCardBtn"
                          >
                            Delete
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui red inverted button storeCardBtn"
                            >
                              Delete
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui red inverted button storeCardBtn"
                          onClick={[Function]}
                        >
                          Delete
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                </div>
              </GridColumn>
            </div>
          </GridRow>
        </StoreCard>
      </withRouter(StoreCard)>
      <withRouter(StoreCard)
        _id="2"
        createdAt="now"
        description="desc"
        dispatch={[Function]}
        imageCount={0}
        key="2"
        state={
          Object {
            "bonusVideoState": Object {
              "bonusVideoFormOpen": false,
              "currentBonusVideoData": Object {
                "_id": "",
                "createdAt": "",
                "description": "",
                "vimeoURL": "",
                "youTubeURL": "",
              },
              "error": null,
              "loadedBonusVideos": Array [],
              "loading": false,
              "responseMsg": "",
            },
            "productState": Object {
              "currentProductData": Object {
                "_id": "",
                "createdAt": "",
                "description": "",
                "details": "",
                "images": Array [],
                "name": "",
                "price": "",
              },
              "error": null,
              "loadedProducts": Array [],
              "loading": false,
              "productFormOpen": false,
              "responseMsg": "",
            },
            "serviceState": Object {
              "currentServiceData": Object {
                "_id": "",
                "createdAt": "",
                "description": "",
                "images": Array [],
                "name": "",
                "price": "",
              },
              "error": null,
              "loadedServices": Array [],
              "loading": false,
              "responseMsg": "",
              "serviceFormOpen": false,
            },
            "storeItemState": Object {
              "currentStoreItemData": Object {
                "_id": "",
                "categories": Array [],
                "createdAt": "",
                "description": "",
                "details": "",
                "images": Array [],
                "name": "",
                "price": "",
                "storeId": "",
                "storeName": "",
              },
              "error": null,
              "loadedStoreItems": Array [],
              "loading": false,
              "numberOfItems": 0,
              "responseMsg": "",
              "storeItemFormOpen": false,
            },
            "storeState": Object {
              "currentStoreData": Object {
                "_id": "",
                "createdAt": "",
                "description": "",
                "images": Array [],
                "title": "",
              },
              "error": null,
              "loadedStores": Array [
                Object {
                  "_id": "1",
                  "createdAt": "now",
                  "description": "desc",
                  "images": Array [],
                  "title": "first",
                },
                Object {
                  "_id": "2",
                  "createdAt": "now",
                  "description": "desc",
                  "images": Array [],
                  "title": "second",
                },
              ],
              "loading": false,
              "responseMsg": "All Ok",
              "storeFormOpen": false,
            },
          }
        }
        title="second"
      >
        <StoreCard
          _id="2"
          createdAt="now"
          description="desc"
          dispatch={[Function]}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "/admin/home/my_stores/manage",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/admin/home/my_stores/manage",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          imageCount={0}
          location={
            Object {
              "hash": "",
              "pathname": "/admin/home/my_stores/manage",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/admin/home/my_stores/manage",
              "url": "/admin/home/my_stores/manage",
            }
          }
          state={
            Object {
              "bonusVideoState": Object {
                "bonusVideoFormOpen": false,
                "currentBonusVideoData": Object {
                  "_id": "",
                  "createdAt": "",
                  "description": "",
                  "vimeoURL": "",
                  "youTubeURL": "",
                },
                "error": null,
                "loadedBonusVideos": Array [],
                "loading": false,
                "responseMsg": "",
              },
              "productState": Object {
                "currentProductData": Object {
                  "_id": "",
                  "createdAt": "",
                  "description": "",
                  "details": "",
                  "images": Array [],
                  "name": "",
                  "price": "",
                },
                "error": null,
                "loadedProducts": Array [],
                "loading": false,
                "productFormOpen": false,
                "responseMsg": "",
              },
              "serviceState": Object {
                "currentServiceData": Object {
                  "_id": "",
                  "createdAt": "",
                  "description": "",
                  "images": Array [],
                  "name": "",
                  "price": "",
                },
                "error": null,
                "loadedServices": Array [],
                "loading": false,
                "responseMsg": "",
                "serviceFormOpen": false,
              },
              "storeItemState": Object {
                "currentStoreItemData": Object {
                  "_id": "",
                  "categories": Array [],
                  "createdAt": "",
                  "description": "",
                  "details": "",
                  "images": Array [],
                  "name": "",
                  "price": "",
                  "storeId": "",
                  "storeName": "",
                },
                "error": null,
                "loadedStoreItems": Array [],
                "loading": false,
                "numberOfItems": 0,
                "responseMsg": "",
                "storeItemFormOpen": false,
              },
              "storeState": Object {
                "currentStoreData": Object {
                  "_id": "",
                  "createdAt": "",
                  "description": "",
                  "images": Array [],
                  "title": "",
                },
                "error": null,
                "loadedStores": Array [
                  Object {
                    "_id": "1",
                    "createdAt": "now",
                    "description": "desc",
                    "images": Array [],
                    "title": "first",
                  },
                  Object {
                    "_id": "2",
                    "createdAt": "now",
                    "description": "desc",
                    "images": Array [],
                    "title": "second",
                  },
                ],
                "loading": false,
                "responseMsg": "All Ok",
                "storeFormOpen": false,
              },
            }
          }
          title="second"
        >
          <GridRow
            style={
              Object {
                "border": "3px solid green",
                "marginTop": "2em",
                "padding": "0.5em",
              }
            }
          >
            <div
              className="row"
              style={
                Object {
                  "border": "3px solid green",
                  "marginTop": "2em",
                  "padding": "0.5em",
                }
              }
            >
              <GridColumn
                computer={12}
                mobile={16}
                style={
                  Object {
                    "border": "2px solid red",
                  }
                }
              >
                <div
                  className="twelve wide computer sixteen wide mobile column"
                  style={
                    Object {
                      "border": "2px solid red",
                    }
                  }
                >
                  <h3>
                    Title: 
                    second
                  </h3>
                  <h5>
                    Description: 
                    desc
                  </h5>
                  <h5>
                    Number of Images: 
                    0
                  </h5>
                  <p>
                    <span>
                      Created At: 
                    </span>
                    NaN-undefined-NaN
                  </p>
                </div>
              </GridColumn>
              <GridColumn
                computer={4}
                mobile={16}
                style={
                  Object {
                    "border": "2px solid blue",
                  }
                }
              >
                <div
                  className="four wide computer sixteen wide mobile column"
                  style={
                    Object {
                      "border": "2px solid blue",
                    }
                  }
                >
                  <Button
                    as="button"
                    className="storeCardBtn"
                    color="green"
                    content="Open"
                    inverted={true}
                    onClick={[Function]}
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui green inverted button storeCardBtn"
                          >
                            Open
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui green inverted button storeCardBtn"
                            >
                              Open
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui green inverted button storeCardBtn"
                          onClick={[Function]}
                        >
                          Open
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                  <Button
                    as="button"
                    className="storeCardBtn"
                    color="orange"
                    content="Edit"
                    inverted={true}
                    onClick={[Function]}
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui orange inverted button storeCardBtn"
                          >
                            Edit
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui orange inverted button storeCardBtn"
                            >
                              Edit
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui orange inverted button storeCardBtn"
                          onClick={[Function]}
                        >
                          Edit
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                  <Button
                    as="button"
                    className="storeCardBtn"
                    color="red"
                    content="Delete"
                    inverted={true}
                    onClick={[Function]}
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui red inverted button storeCardBtn"
                          >
                            Delete
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui red inverted button storeCardBtn"
                            >
                              Delete
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui red inverted button storeCardBtn"
                          onClick={[Function]}
                        >
                          Delete
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                </div>
              </GridColumn>
            </div>
          </GridRow>
        </StoreCard>
      </withRouter(StoreCard)>
    </Route>
  </div>
</Grid>
`;
